crates/ruff_server/src/session/settings.rs

```diff
@@ -31,6 +31,7 @@ pub(crate) struct ResolvedClientSettings {
 #[derive(Clone, Debug, Default)]
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub(crate) struct ResolvedEditorSettings {
+    pub(super) configuration: Option<PathBuf>,
```

Comment:
I'm open to using a better name for this 😅

Reply:
I think we'll need to expand tildes and environment variables for settings that users provide in VS Code. (We do this within Ruff when you pass `--config`.) Should that happen within the LSP?

Reply:
I would expect this to be a VS Code extension functionality, except if we support it for `ruff check --config` too.

Reply:
Environment variable resolution is definitely something we should have, but I think that's beyond the scope of this PR. As for tilde resolution - shouldn't that be resolved already in `parse_pyproject_toml`?

Reply:
Okay, I did some testing and it looks like tilde expansion isn't handled by the standard library automatically. I'll open a follow-up PR that uses `shellexpand` to expand tildes and environment variables.